smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/ShapeValueGenerator.kt

```diff
@@ -211,14 +227,37 @@ class ShapeValueGenerator(
         }
 
         override fun arrayNode(node: ArrayNode) {
-            val memberShape = generator.model.expectShape((currShape as CollectionShape).member.target)
-            var i = 0
-            node.elements.forEach { element ->
-                generator.writeShapeValueInline(writer, memberShape, element)
-                if (i < node.elements.size - 1) {
-                    writer.writeInline(",\n")
+            when (currShape.type) {
+                ShapeType.DOCUMENT -> {
+                    writer
+                        .writeInline("#T(\n", RuntimeTypes.Core.Smithy.Document)
+                        .indent()
+                        .writeInline("listOf(\n")
+                        .indent()
+
+                    node.elements.forEach {
+                        generator.writeShapeValueInline(writer, currShape, it)
+                        writer.writeInline(",\n")
+                    }
+
+                    writer
+                        .dedent()
+                        .writeInline(")\n")
+                        .dedent()
+                        .writeInline(")")
```

Comment:
**Style**: The manual indenting/dedenting is unnecessary here:

```kotlin
writer.withBlock("#T(", ")", RuntimeTypes.Core.Smithy.Document) {
    withBlock("listOf(", ")") {
        node.elements.forEach {
            generator.writeShapeValueInline(writer, currShape, it)
            writeInline(",\n")
        }
    }
}
```

Reply:
So this was deliberate -- I had `writeBlock`ified at first, but it doesn't generate output in the spirit of the method `writeShapeValueInline` - keyword being "inline", you end up with some extra line breaks. The code is still functional but not necessarily reader-friendly and the output isn't truly inline at that point.

It doesn't super matter to me, especially considering that this `ShapeValueGenerator` is effectively only generating for test code, but I suppose it could be otherwise in the future.

We could also expose `withInlineBlock` on the writer, because the manual *denting is indeed nasty.

At the very least, if we do keep it, a comment might be warranted.

Reply:
Oh I see, I missed that the last write doesn't want a newline. `withInlineBlock` is a great idea and might lead to a simplification of this new block and probably some existing blocks as well. Up to you if you want to tackle it in this PR. Otherwise, my concern is addressed.